src/assemble_workflow/bundle_recorder.py

```diff
@@ -6,60 +6,58 @@
 
 import os
 from urllib.parse import urljoin
-
 from manifests.bundle_manifest import BundleManifest
-
+from assemble_workflow.public_urls import PublicURLpath
 
 class BundleRecorder:
     def __init__(self, build, output_dir, artifacts_dir):
         self.output_dir = output_dir
         self.build_id = build.id
-        self.public_url = os.getenv("PUBLIC_ARTIFACT_URL", None)
         self.version = build.version
         self.tar_name = self.__get_tar_name(build)
         self.artifacts_dir = artifacts_dir
         self.arch = build.architecture
+        self.urls = PublicURLpath(build).urls_path
         self.bundle_manifest = self.BundleManifestBuilder(
             build.id,
             build.name,
             build.version,
             build.architecture,
-            self.__get_tar_location(),
+            self.__get_tar_location(self.urls["bundles"]),
         )
 
     def __get_tar_name(self, build):
         parts = [build.name.lower().replace(" ", "-"), build.version, "linux", build.architecture]
         return "-".join(parts) + ".tar.gz"
 
-    def __get_public_url_path(self, folder, rel_path):
-        path = "/".join((folder, self.version, self.build_id, self.arch, rel_path))
-        return urljoin(self.public_url + "/", path)
+    def __get_public_url_path(self, url_path, rel_path):
+        return urljoin(url_path + "/", rel_path)
 
-    def __get_location(self, folder_name, rel_path, abs_path):
-        if self.public_url:
-            return self.__get_public_url_path(folder_name, rel_path)
+    def __get_location(self, url_path, rel_path, abs_path):
```

Comment:
Don't use a parameter, have it as a class variable, the base url should be passed in during construction of the bundle recorder since it should be the same value for every bundled artifact